app.json

```diff
@@ -0,0 +1,279 @@
+{
+  "name": "spoke",
+  "description": "Spoke",
+  "repository":"https://github.com/moveonorg/spoke",
+  "success_url":"/setup.html",
+  "buildpacks": [
+      { "url": "https://github.com/heroku/heroku-buildpack-nodejs"}
+  ],
+  "env": {
+    "NODE_ENV": {
+      "description": "node environment determines the application build/start process",
+      "required": true,
+      "value": "production"
+    },
+
+    "SUPPRESS_SELF_INVITE": {
+      "description": "if you want to suppress an organizational invite creation when logging in, set this environment to 1",
+      "required": true,
+      "value": ""
+    },
+
+    "SUPPRESS_MIGRATIONS": {
+      "description": "to suppress table migration on server start up",
+      "required": false,
+      "value": ""
+    },
+
+    "JOBS_SAME_PROCESS": {
+      "description": "",
+      "required": true,
+      "value": 1
+    },
+
+    "AWS_ACCESS_KEY_ID": {
+      "description": "in order to export data to an s3 bucket",
+      "required": true,
+      "value": ""
+    },
+
+    "AWS_SECRET_ACCESS_KEY": {
+      "description": "in order to export data to an s3 bucket",
+      "required": true,
+      "value": ""
+    },
+
+    "AWS_S3_BUCKET_NAME": {
+      "description": "in order to export data to an s3 bucket",
+      "required": true,
+      "value": ""
+    },
+
+    "APOLLO_OPTICS_KEY": {
+      "description": "for apollo optics integrat",
+      "required": false,
+      "value": ""
+    },
+
+    "DEV_APP_PORT": {
+      "description": "node port",
+      "required": true,
+      "value": "8090"
+    },
+
+    "OUTPUT_DIR": {
+      "description": "for production build placement",
+      "required": true,
+      "value": "./build"
+    },
+
+    "PUBLIC_DIR": {
+      "description": "for production build placement",
+      "required": true,
+      "value": "./build/client"
+    },
+
+    "ASSETS_DIR": {
+      "description": "for assets directory in production build placement",
+      "required": true,
+      "value": "./build/client/assets"
+    },
+
+    "ASSETS_MAP_FILE": {
+      "description": "for assets directory placement",
+      "required": true,
+      "value": "assets.json"
+    },
+
+    "CAMPAIGN_ID": {
+      "description": "",
+      "required": false,
+      "value": "campaign-id-hash"
+    },
+
+    "DB_HOST": {
```

Comment:
I think all of these can be setup automatically by Heroku without the user needing to enter them.
We would probably need something like this in the src/server/models/thinky.js file:
```
+} else if (process.env.DATABASE_URL) {
+  var databaseType = process.env.DATABASE_URL.match(/^\w+/)[0]
+  config = {
+    client: (/postgres/.test(databaseType) ? 'pg' : databaseType),
+    connection: process.env.DATABASE_URL,
+    pool: {
+      min: process.env.DB_MIN_POOL || 2,
+      max: process.env.DB_MAX_POOL || 10
+    }
+  }
```

Reply:
db_host  and db_ creds are derived from heroku database credentials page so i have to figure out a way to populate these values from the other page

Reply:
See here, for example:
https://devcenter.heroku.com/changelog-items/438